src/ray/common/client_connection.cc

```diff
@@ -7,20 +7,59 @@
 
 namespace ray {
 
+ray::Status TcpConnect(boost::asio::ip::tcp::socket &socket,
+                       const std::string &ip_address_string, int port) {
+  boost::asio::ip::address ip_address =
+      boost::asio::ip::address::from_string(ip_address_string);
+  boost::asio::ip::tcp::endpoint endpoint(ip_address, port);
+  boost::system::error_code error;
+  socket.connect(endpoint, error);
+  if (error) {
+    return ray::Status::IOError(error.message());
+  } else {
+    return ray::Status::OK();
+  }
+}
+
+template <class T>
+ServerConnection<T>::ServerConnection(boost::asio::basic_stream_socket<T> &&socket)
+    : socket_(std::move(socket)) {}
+
+template <class T>
+ray::Status ServerConnection<T>::WriteMessage(int64_t type, size_t length,
+                                              const uint8_t *message) {
+  std::vector<boost::asio::const_buffer> message_buffers;
+  auto write_version = RayConfig::instance().ray_protocol_version();
+  message_buffers.push_back(boost::asio::buffer(&write_version, sizeof(write_version)));
+  message_buffers.push_back(boost::asio::buffer(&type, sizeof(type)));
+  message_buffers.push_back(boost::asio::buffer(&length, sizeof(length)));
```

Comment:
We probably want `length` to be an `int64_t` so it has a guaranteed size, right? If two machines compile `size_t` to have different sizes, then things will break.

Reply:
This caused problems before which I had to fix in e.g., https://github.com/apache/arrow/commit/81319d9c2784a626af318911c183701d320888f4.

Reply:
@stephanie-wang

Reply:
@atumanov? We get a parameter type mismatch here if we use `int64_t`, but happy to make the change if it make sense.

Reply:
This is absolutely critical.. Make it `int64_t` and read into `int64_t`, then you can static cast to `size_t`.